### ISSUE TITLE
build: cypress build no longer optional?

### ISSUE THREAD

**Issue by davepacheco** [cockroachdb/cockroach#93162]
**Describe the problem**

I saw the 22.2 release announcement (congrats!) and went to build an OSS build on illumos.  I ran into this failure:

```
error /home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach/pkg/ui/workspaces/e2e-tests/node_modules/cypress: Command failed.
Exit code: 1
Command: node index.js --exec install
Arguments: 
Directory: /home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach/pkg/ui/workspaces/e2e-tests/node_modules/cypress
Output:
Platform: "sunos" is not supported.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
gmake: *** [Makefile:371: pkg/ui/yarn.installed] Error 1
gmake: *** Waiting for unfinished jobs....
```

Is this a regression of #47694?

I applied this patch:

```
index c93a67e..be5518c 100755
--- a/src/github.com/cockroachdb/cockroach/pkg/ui/not-yarn-workspace.sh
+++ b/src/github.com/cockroachdb/cockroach/pkg/ui/not-yarn-workspace.sh
@@ -7,7 +7,7 @@ set -euo pipefail
   yarn --cwd workspaces/db-console/src/js install
   yarn --cwd workspaces/cluster-ui install
   yarn --cwd workspaces/db-console install
-  yarn --cwd workspaces/e2e-tests install
+  # yarn --cwd workspaces/e2e-tests install
 )
 
 cat << "EOF"
```

and was able to complete a build.

Let me know if there's more information you need.  I removed the other template sections from this report because they seemed not relevant.)

Jira issue: CRDB-22195

**Comment by knz** [comment]
@sjbarag @rickystewart can you have a look?

**Comment by davepacheco** [comment]
Any update on this one?

**Comment by sjbarag** [comment]
Sorry @davepacheco , I hadn't seen this thread until now!  Agreed, this does look like a regression of https://github.com/cockroachdb/cockroach/issues/47694 — let me see what I can do, both as a short-term workaround (without you having to manually patch that file) and as a longer-term real solution.

So I know if I'm reproing properly, are you building with `make`, `build/builder.sh`, or `bazel`?

**Comment by sjbarag** [comment]
This impacts `freebsd` as well, for clarity: https://github.com/cockroachdb/cockroach/issues/87369

**Comment by knz** [comment]
@sjbarag this just got fixed by #93800 right?

**Comment by davepacheco** [comment]
> So I know if I'm reproing properly, are you building with `make`, `build/builder.sh`, or `bazel`? (That's a silly-ish question — that `not-yarn-workspace.sh` was involved at all implies not a Bazel build, and `platform: "sunos"` implies not-docker IIRC. But still, couldn't hurt to confirm I suppose.)

Thanks. We’re using make.

**Comment by sjbarag** [comment]
> @sjbarag this just got fixed by #93800 right?

In master, yes! And fixed with #93803 in 22.2.

**Comment by sjbarag** [comment]
@davepacheco we should be good to go know :) I can get you a date for the next release when I get back to not-my-phone, but for the time being the tip of the `release-22.2` branch should build for you!
I’m going to close this, but if something comes up please let me know :)

Thanks again for calling this out!

**Comment by davepacheco** [comment]
Thanks!

**Comment by davepacheco** [comment]
It looks like #93803 still hasn't landed in 22.2.8?
